cookbooks/bcpc-hadoop/recipes/graphite_to_zabbix.rb

```diff
@@ -234,47 +242,55 @@
 
           # For all triggers, a companion trigger is created to check whether
           # the zabbix sender cron job is active and sends data to Zabbix.
-          #cron_check_cond << "{" + "#{trigger_host}" + ":" + trigger_key +
-          #  ".nodata(#{node["bcpc"]["hadoop"]["zabbix"]["cron_check_time"]})}=1"
+          #cron_check_cond << '{' + "#{trigger_host}" + ':' + trigger_key +
+          #  ".nodata(#{node['bcpc']['hadoop']['zabbix']['cron_check_time']})}=1"
         else
           trigger_info[:triggerid] = trigger_id
           updateTriggersArr.push(trigger_info)
-        end # End of "if (trigger_id = existing_triggers[trigger_name]).nil?"
+        end # End of 'if (trigger_id = existing_triggers[trigger_name]).nil?'
 
         # Create/Update Actions
         action_status = node[:bcpc][:hadoop][:zabbix][:enable_alarming] ? status : 1
         esc_period = attrs['esc_period'].nil? ? node[:bcpc][:hadoop][:zabbix][:escalation_period] : attrs['esc_period']
 
         action_info = {
-          'status' => action_status, 'esc_period' => esc_period,
-          'filter'=> {  
-            'evaltype' => 1,
+          'status' => action_status,
+          'esc_period' => esc_period,
+          'filter'=> {
+            'evaltype' => default_action['filter']['evaltype'],
             'conditions' => [
-              {"conditiontype" => 3,"operator" => 2,"value" => trigger_name},
-              {"conditiontype" => 5,"operator" => 0,"value" => 1},
-              {"conditiontype" => 16,"operator" => 7, "value" => ''}
+              {
+                'conditiontype' => default_action['filter']['conditions']['conditiontype'],
+                'operator' => default_action['filter']['conditions']['operator'],
+                'value' => trigger_name
+              },
+              node.default['bcpc']['zabbix']['action']['filter']['conditions']['trigger_problem'],
+              node.default['bcpc']['zabbix']['action']['filter']['conditions']['maintenance_off']
             ]
           },
           'operations' => [{
-            "operationtype" => 1, "esc_step_from" => 2, "esc_step_to" => 2,
-            "opcommand" => {
-              "command" => "#{node['bcpc']['zabbix']['scripts']['mail']}" +
+            'operationtype' => default_action['operations']['operationtype'],
+            'esc_step_from' => default_action['operations']['esc_step_from'],
+            'esc_step_to' => default_action['operations']['esc_step_to'],
```

Comment:
for each zabbix action, a series of escalation steps can be defined. (https://www.zabbix.com/documentation/2.4/manual/config/notifications/action/escalations)